### ISSUE TITLE
Problem matcher issue

### ISSUE THREAD

**Issue by jonasbn** [nosborn/github-action-markdown-cli#26]
Hello,

I am a very happy user of the Markdownlint GitHub Action. The latest release 3.0.0 does however give me some problems, since all of my PRs from @dependabot fail their builds with the following error:

```text
Error: Unable to process command '::add-matcher::.github/markdownlint-problem-matcher.json' successfully.
Error: Could not find file '/home/runner/work/ebirah/ebirah/.github/markdownlint-problem-matcher.json'.
``` 

Problem matchers are awesome and I am welcoming the addition, but the implementation does not seem to take into consideration, that the problem matcher is not available to the consumers of the GitHub Action.

I attempted to add the [referenced file](https://github.com/nosborn/github-action-markdown-cli/blob/master/.github/markdownlint-problem-matcher.json) one of my repositories and it worked

```json
{
  "problemMatcher": [
    {
      "owner": "markdownlint",
      "pattern": [
        {
          "code": 3,
          "file": 1,
          "line": 2,
          "message": 4,
          "regexp": "^(.+):(\\d+)(?:\\d+)? (\\S+) (.+)$"
        }
      ]
    }
  ]
}
```

The problem seem to be this line:

```shell
echo '::add-matcher::.github/markdownlint-problem-matcher.json'
```

This file is not available in consuming repositories and the documentation does not describe that it should be added. 

I would love to write a PR with an update to the documentation, but before doing so I would rather point you to the problem matcher I use with a `xmllint` based GitHub Action. The configuration looks as follows:

```yaml
name: XMLlint Action
on: push

jobs:
  build:
    name: XMLlint
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@master
    - uses: korelstar/xmllint-problem-matcher@v1
    # This step references the directory that contains the action.
    - uses: ./.github/actions/xmllint
```

You can find [the file here](https://github.com/DK-Hostmaster/epp-xsd-files/blob/master/.github/workflows/xmllint.yml)

It references the external problem matcher and simply uses it. The problem matcher lives in its [own repository](https://github.com/korelstar/xmllint-problem-matcher).

The interesting part comes from it's documentation:

> Add the step to your workflow, before xmllint is called.

Meaning this line:

```yaml
    - uses: korelstar/xmllint-problem-matcher@v1
```

Should come before this one:

```yaml
    - uses: ./.github/actions/xmllint
```

The beauty is in the composition.

Please have a look at how it was solves there, since it might be an inspiration. Else I guess the key is in this in the  "korelstar / xmllint-problem-matcher" implementation:

```javascript
console.log(`::add-matcher::${__dirname}/xmllint-matcher.json`);
```

I am by no means an expert on problem matcher, but I will gladly help if I can, also because I am a heavy user of this GitHub Action.

As I stated I got it to work for [one of my repositories](https://github.com/jonasbn/bash_completion_perlbrew), but I would prefer not that have to maintain the problem matcher in all of my repositories, since it is more closely bound to the actual action that to the consumers.

Please let me know if you need additional information from me - take care and stay safe and thanks for the awesome action.

**Comment by nosborn** [comment]
Thanks for the detailed issue @jonasbn. This should be fixed in v3.0.1.

The problem matcher is intended to work out-of-the-box without any changes to the calling workflow and replaces code in the action that previously commented on PRs. Unfortunately, I did this in a manner that only worked for the tests in *this* repository and so, as you observed, it wasn't available to consumers of the action. 🤦 

Apologies for the bad release.

**Comment by jonasbn** [comment]
I have just confirmed that 3.0.1 works in one of my repositories. I expect to hear from @dependabot one of the coming days :-)
